packages/x-data-grid/src/models/gridRows.ts

```diff
@@ -116,9 +116,9 @@ export interface GridDataGroupNode extends GridBasicGroupNode {
 
 export interface GridDataSourceGroupNode extends GridDataGroupNode {
   /**
-   * If true, this node has children on server.
+   * Number of children this node has on the server. Returns `-1` if unknown.
```

Comment:
I think `-1` means "non-zero", not just "unknown".

Reply:
cc @MBilalShafi could confirm

Reply:
Yes, `-1` would mean there _are_ some children but the count is unknown.